components/IngredientDetails.js

```diff
@@ -0,0 +1,132 @@
+import { useRouter } from "next/router";
+import Image from "next/image";
+import styled from "styled-components";
+import ingredients from "@/assets/ingredients.json";
+import { useEffect, useState } from "react";
+
+const IngredientDetails = () => {
+  const flavorColors = {
+    Bitter: "#D2691E",
+    Creamy: "#FFFACD",
+    Fresh: "#98FB98",
+    Herbal: "#228B22",
+    Pungent: "#FF6347",
+    Salty: "#F0E68C",
+    Savory: "#D2B48C",
+    Spicy: "#cf2210",
+    Sour: "#d9df0c",
+    Sweet: "#FFB6C1",
+    Tangy: "#FFA07A",
+    Umami: "#188989",
+  };
+
+  const router = useRouter();
+  const { id } = router.query;
+  const [ingredient, setIngredient] = useState(null);
+
+  useEffect(() => {
+    if (id) {
+      const foundIngredient = ingredients.find(
+        (ingredient) => ingredient._id === id
+      );
+      setIngredient(foundIngredient);
+    }
+  }, [id]);
+
+  if (!ingredient) {
+    return <p>Loading...</p>;
+  }
+
+  return (
+    <>
+      <BackLink href="/">← Back</BackLink>
+      <Container>
+        <ImageWrapper>
+          <StyledImage
+            src={ingredient.imgUrl}
+            alt={ingredient.name}
+            fill
+            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
+          />
+        </ImageWrapper>
+        <StyledContent>
+          <Name>{ingredient.name}</Name>
+          <br />
+          <Flavor $color={flavorColors[ingredient.flavor]}>
+            #{ingredient.flavor}
+          </Flavor>
+        </StyledContent>
```

Comment:
This code exists exactly the same way in the IngredientsList component and it is not the best practise to repeat yourselves.

Why not create a separate component that returns this JSX? 
This new component could contain all the necessary styling as well as the flavorColors object and be called in the IngredientsList component as well as the IngredientDetails component.

Reply:
We have different styling in IngredientDetails and IngredientList. is it possible to style this one component differntly?

Reply:
@NikRoe We managed to create a new component for our ImageWrapper and StyledContent but somehow, in the IngrediendsList the pictures it's not being rendered, while in the IngredientsDetail it is.

Reply:
Hi there,

this is not really much information to work with. 

Please push the current status of your branch and I will take a look at what is missing / causing the error.